planet/clients/orders.py

```diff
@@ -318,8 +316,14 @@ async def download_order(self,
     @staticmethod
     def _get_order_locations(order):
         links = order['_links']
-        results = links.get('results', None)
-        return list(r['location'] for r in results if r)
+        results = links.get('results', [])
+        try:
+            return list(r['location'] for r in results if r)
+        except TypeError:
```

Comment:
wow! so this still throws a TypeError even with the change to line 319 and the `...if r` check? That's a surprise to me!